docs/infrastructure/EKS/main/cluster_masters.tf

```diff
@@ -30,11 +30,46 @@ resource "aws_eks_cluster" "main" {
 
 }
 
+# Get an authentication token to communicate with an EKS cluster, using the "kubernetes" provider.
+data "aws_eks_cluster_auth" "eks_cluster_auth" {
+  name = "${aws_eks_cluster.main.name}"
+}
+
+provider "kubernetes" {
```

Comment:
Please move provider block to the `config.tf` file. This way it is easier to keep track of where everything is coming from.